src/mqt/qmap/visualization/search_visualizer.py

```diff
@@ -0,0 +1,245 @@
+"""A class handling data logging for a search process and providing methods to visualize that data."""
+
+from __future__ import annotations
+
+from tempfile import TemporaryDirectory
+from typing import TYPE_CHECKING, Callable, Literal, MutableMapping
+
+if TYPE_CHECKING:
+    import types
+
+    from typing_extensions import Self
+
+from .visualize_search_graph import SearchNode, visualize_search_graph
+
+if TYPE_CHECKING:
+    from ipywidgets import Widget
+
+
+class SearchVisualizer:
+    """Handling data logging for a search process and providing methods to visualize that data."""
+
+    def __init__(self, data_logging_path: str | None = None) -> None:
+        """Handling data logging for a search process and providing methods to visualize that data.
+
+        Args:
+            data_logging_path: Path to an empty directory, in which the search process should log all data.
+                Defaults to None, in which case a temporary folder will be created.
+        """
+        if data_logging_path is not None:
+            self.data_logging_path: str | None = data_logging_path
```

Comment:
Because of the `data_logging_path is not None` check above that line, mypy actually infers only the type `str`, but when the visualizer is closed `self.data_logging_path` is later set to `None`, so the wider type `str | None` is, in fact, required and seems to need the explicit type hint for mypy.

Reply:
Ah. Could have figured that it's something like that. Mypy is strange sometimes 🙃